### ISSUE TITLE
Local Vagrant environment for portable integration testing

### ISSUE THREAD

**Issue by acornies-tc** [hashicorp/faas-nomad#43]
Since this feature doesn't quite fit the issue template, I'll outline it below:

Provide a portable Vagrant environment via Vagrantfile that can:
- Provision entire Hashicorp stack (Docker, Consul, Nomad, Vault) in a standalone VM
- Run OpenFaaS components using Nomad w/ Docker
- Add debugging launch configuration which connects faas gateway to faas-nomad
- Assume Go development is done on host/workstation, the rest is provisioned in the VM
- Facilitate integration testing for all supported features of OpenFaaS

Discussed briefly with @nicholasjackson, but wanted to track it.

**Comment by acornies** [comment]
This is me ^ forgot to switch accounts when drafting up the issue.

**Comment by alexellis** [comment]
This would be useful for testing. Do you know if it'd work with KVM on Linux too?

**Comment by acornies** [comment]
Hi @alexellis, in theory it should be fine with vagrant-libvirt plugin and the libvirt provider (I haven't tried it myself). More info here: https://github.com/vagrant-libvirt/vagrant-libvirt. 

I could see adding something like this in the Vagrantfile:
```ruby
config.vm.provider "libvirt" do |lv|
  lv.cpus = 4
  lv.memory = 1024
  ...
end
```
then to trigger it would be: `vagrant up --provider=libvirt`

**Comment by nicholasjackson** [comment]
Liking this so far, I have left a couple of comments, I think we can probably make things a little more generic.

**Comment by nicholasjackson** [comment]
Thanks @acornies this is merged in #46
